### PR TITLE
Added withUrlEncodedBody and helper functions

### PR DESCRIPTION
**Pull request by Istar-Eldritch**
Please review the code and give me feedback if you think is necessary to change something.

### CODE REVIEW

File: src/Http/Extra.elm [hunk 1]
```diff
@@ -204,6 +205,15 @@ withMultipartStringBody =
     >> withMultipartBody
 
 
+{-| Convenience function for adding url encoded bodies
+  post "https://example.com/api/whatever"
+    |> withUrlEncodedBody [("user", "Evan"), ("pwd", "secret")]
+-}
+withUrlEncodedBody : List (String, String) -> RequestBuilder -> RequestBuilder
+withUrlEncodedBody body =
+  withBody <| Http.string <| joinUrlEncoded <| body
```
Comment (lukewestby) [inline src/Http/Extra.elm]: can we use `>>` here instead?

``` elm
withUrlEncodedBody =
  joinUrlEncoded >> withStringBody
```

File: src/Http/Extra.elm [hunk 2]
```diff
@@ -392,3 +402,25 @@ testing
...
+
+
+queryEscape : String -> String
+queryEscape string =
+  string |> Http.uriEncode |> replace "%20" "+"
+
+
+replace : String -> String -> String -> String
+replace old new string =
+  string |> String.split old |> String.join new
```
Comment (lukewestby) [inline src/Http/Extra.elm]: This is clever! Much simpler than using `Regex` :smile:

**Comment by lukewestby** [comment]
@Istar-Eldritch this is great, thank you! I just had the one suggestion.

**Comment by Istar-Eldritch** [comment]
Hey @lukewestby

Was not aware of the composition operator. Maybe it's less declarative but makes it way cleaner.

I modified also the helper functions to make use of it.

**Comment by lukewestby** [comment]
cool! thanks for making the update. i love `>>` and `<<`, they're especially useful when used in combination with `|>`. :+1: this looks good, i'm gonna merge it and publish 5.1.0

File: src/Http/Extra.elm (hunk 2 above)
Comment (Istar-Eldritch) [inline src/Http/Extra.elm]: Stole from the original Evan's code with sligly modifications. I don't really deserve credit on this stuff. ;)

**Comment by lukewestby** [comment]
published 5.1.0!

**Pull request merged**